README.md

```diff
@@ -1,31 +1,63 @@
 # esp-println
 
-Provides `print!`, `println!` and `dbg!` implementations for various Espressif devices.
-
-- Supports ESP32, ESP32-C2/C3/C6, ESP32-H2, ESP32-S2/S3, and ESP8266
-- Dependency free (not even depending on `esp-hal`, one optional dependency is `log`, another is `critical-section`)
-- Supports JTAG-Serial output where available
-- Supports RTT (lacking working RTT hosts besides _probe-rs_ for ESP32-C3 and ESP32-C6)
-- `no-op` features turns printing into a no-op
-
-## Usage
-
-In your `Cargo.toml`, under `[dependencies]`, add:
-
-```
-cargo add esp-println --features esp32
+A library that provides `print!`, `println!`, `dbg!` implementations and
+logging capabilities for Espressif devices.
+- Supports all Espressif devices.
+- Supports different communication methods:
+    - UART (Default)
+    - JTAG-Serial (Only available in ESP32-C3, ESP32-C6, ESP32-H2, ESP32-S3)
+    - [RTT]
+    - No-op: Turns printing into a no-op
+- Supports [`defmt`] backend
+
+# Usage
+
+```toml
+esp-println = { version = "0.7.0", features = ["esp32c2"] }
 ```
-
-Replace `esp32` with your target, if necessary.
+or `cargo add esp-println --features esp32c2`
+It's important to specify your target device as feature.
 
 Then in your program:
 
-```
+```rust
 use esp_println::println;
 ```
 
 You can now `println!("Hello world")` as usual.
 
+# Features
+
+- There is one feature for each supported target: `esp32`, `esp32c2`,
+  `esp32c3`, `esp32c6`, `esp32h2`, `esp32s2`, `esp32s3` and `esp8266`.
+   - One of these features must be enabled.
+   - Only one of these features can be enabled at a time.
+- There is one feature for each supported communication method: `uart`,
+  `jtag-serial` and `rtt` `no-op`.
+    - Only one of these features can be enabled at a time.
+- `log`: Enables logging using [`log` crate].
+- `colors` enable colored logging.
+   - Only effective when using the `log` feature.
+- `critical-section` enables critical sections.
+- There are two ways to use this library with [`defmt`]:
+   - `defmt-espflash`: This is intended to be used with [`espflash`], see `-L/--log-format` argument of `flash` or `monitor` subcommands of `espflash` and `cargo-espflash`.
+     Uses [rzCOBS] encoding and adds framing.
+   - `defmt-raw`: Raw defmt output without additional framing. ⚠️ Be careful when using this feature: you must only write
+     output using `defmt` macros, or you may irrecoverably corrupt the output stream! This means that even the bootloader's output
+     must be disabled.
+
+   `defmt` features can also be used with [`probe-rs`].
+
+[`probe-rs`]: https://probe.rs/
+
+## Default Features
+
+By default, we use the `uart`, `critial-section` and `colors` features.
```

Comment:
If the `colors` feature is enabled by default, and the `log` feature is required and _not_ a default feature, is this not problematic?

Reply:
no - without the log feature colors is no-op

Reply:
The build script will panic if `colors` is enabled and `log` is not in its current state:
https://github.com/esp-rs/esp-println/pull/68/files#diff-d0d98998092552a1d3259338c2c71e118a5b8343dd4703c0c7f552ada7f9cb42R47-R50

Reply:
ah lol - yeah introduced in this PR

probably we shouldn't have that checked or at least lower it to a warning